### PR TITLE
thor: Add helGetCurrentCpu syscall

### PR DESCRIPTION
**Pull request by Qwinci**

### CODE REVIEW

**Review by no92 — APPROVED** [review]
LGTM

File: hel/include/hel.h [hunk 1]
```diff
@@ -929,6 +930,8 @@ HEL_C_LINKAGE HelError helWriteGsBase(void *pointer);
 
 HEL_C_LINKAGE HelError helReadGsBase(void **pointer);
 
+HEL_C_LINKAGE HelError helGetCurrentCpu(int *index);
```
Comment (qookei) [inline hel/include/hel.h:933]: Could you document this?

File: kernel/thor/generic/main.cpp [hunk 2]
```diff
@@ -876,6 +876,9 @@ void handleSyscall(SyscallImageAccessor image) {
 	case kHelCallSetAffinity: {
 		*image.error() = helSetAffinity((HelHandle)arg0, (uint8_t *)arg1, (size_t)arg2);
 	} break;
+	case kHelCallGetCurrentCpu: {
+		*image.error() = helGetCurrentCpu((int*)arg0);
```
Comment (ElectrodeYT) [inline kernel/thor/generic/main.cpp:880]: This should not use output paramters; this should use `*image.out{}`

**Review by ElectrodeYT — CHANGES_REQUESTED** [review]
One thing

File: kernel/thor/generic/main.cpp (hunk 2 above)
Comment (Qwinci) [reply kernel/thor/generic/main.cpp]: I changed it to use out0. And I assume helSyscall0_1 is the correct one to use in hel-syscalls.h for that?

Comment (ElectrodeYT) [reply kernel/thor/generic/main.cpp]: Yep, it should be. However, you should still be returning a HelError in `*image.error()`. Returning `kHelErrNone` unconditionally should suffice.

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**